client/chat/channel-header.tsx

```diff
@@ -0,0 +1,252 @@
+import React, { useMemo } from 'react'
+import { useTranslation } from 'react-i18next'
+import styled from 'styled-components'
+import { ReadonlyDeep } from 'type-fest'
+import { assertUnreachable } from '../../common/assert-unreachable'
+import {
+  BasicChannelInfo,
+  DetailedChannelInfo,
+  JoinedChannelInfo,
+  SbChannelId,
+} from '../../common/chat'
+import { CAN_LEAVE_SHIELDBATTERY_CHANNEL } from '../../common/flags'
+import { matchLinks } from '../../common/text/links'
+import { useOverflowingElement } from '../dom/overflowing-element'
+import { MaterialIcon } from '../icons/material/material-icon'
+import { IconButton } from '../material/button'
+import { DestructiveMenuItem } from '../material/menu/item'
+import { MenuList } from '../material/menu/menu'
+import { Popover, useAnchorPosition, usePopoverController } from '../material/popover'
+import { shadow2dp } from '../material/shadows'
+import { Tooltip, TooltipContent } from '../material/tooltip'
+import { ExternalLink } from '../navigation/external-link'
+import { LoadingDotsArea } from '../progress/dots'
+import { useStableCallback } from '../state-hooks'
+import { background700, colorTextFaint, colorTextSecondary } from '../styles/colors'
+import { caption, headline6, singleLine } from '../styles/typography'
+import { ChannelBadge } from './channel-badge'
+
+const ChannelHeaderRoot = styled.div<{ $hasRightPadding: boolean }>`
```

Comment:
This isn't a good name for a prop, it should correspond to *why* this value differs, not *what* the value changes